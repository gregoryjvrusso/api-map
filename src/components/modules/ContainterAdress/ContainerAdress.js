import React from 'react'
import styled from 'styled-components'
import Text from '../../elements/Text/Text'

const Container = styled.div`
  width: 80%;
  padding: 0 1em;
  display: flex;
  flex-direction: column;
`

const ContainerAdress = ({ data }) => {
  return (
    <Container>
      <Text
        text={data.logradouro}
        fontSize='2em'
        fontWeight='bold'
        margin='0'
      />
      <Text text={data.bairro} margin='1em 0 0' />
      <Text text={`${data.localidade}-${data.uf}`} margin='1em 0 0' />
      <Text text={data.cep} margin='1em 0 1em' />
    </Container>
  )
}

export default ContainerAdress