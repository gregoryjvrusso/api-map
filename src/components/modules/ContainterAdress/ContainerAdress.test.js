import React from 'react'
import { render } from '@testing-library/react'
import ContainerAdress from './ContainerAdress.js'

const mock = {
    "cep": "01001-000",
    "logradouro": "Praça da Sé",
    "complemento": "lado ímpar",
    "bairro": "Sé",
    "localidade": "São Paulo",
    "uf": "SP",
    "unidade": "",
    "ibge": "3550308",
    "gia": "1004"
  }
describe('Container Adress component', () => {
  it('deve conseguir renderizar [snapshot]', async () => {
    const { container } = render(<ContainerAdress data={mock} />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
