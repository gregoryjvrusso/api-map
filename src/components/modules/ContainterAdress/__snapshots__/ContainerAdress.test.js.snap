// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container Adress component deve conseguir renderizar [snapshot] 1`] = `
<div
  class="sc-bwzfXH wqKLy"
>
  <p
    class="sc-bdVaJa fTxPHd"
    font-size="2em"
    font-weight="bold"
  >
    Praça da Sé
  </p>
  <p
    class="sc-bdVaJa ffYDP"
  >
    Sé
  </p>
  <p
    class="sc-bdVaJa ffYDP"
  >
    São Paulo-SP
  </p>
  <p
    class="sc-bdVaJa dqUOud"
  >
    01001-000
  </p>
</div>
`;
